api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -734,3 +737,15 @@ message CustomInlineHeader {
   // The type of the header that is expected to be set as the inline header.
   InlineHeaderType inline_header_type = 2 [(validate.rules).enum = {defined_only: true}];
 }
+
+message MemoryAllocatorManager {
+  // Configures tcmalloc to perform background release of free memory in amount of bytes per ``memory_release_interval`` interval.
+  // If equals to ``0``, no memory release will occur. Defaults to ``0``.
+  // Currently memory releasing for gperf tcmalloc is not supported.
+  uint64 bytes_to_release = 1 [(validate.rules).uint64 = {gte: 1}];
+
+  // Interval in milliseconds for memory releasing. If specified, during every
+  // interval Envoy will try to ``release bytes_to_release`` of free memory back to operating system for reuse.
+  // Defaults to 1000 milliseconds.
+  google.protobuf.Duration memory_release_interval = 2;
+}
```

Comment:
If we think we will never support gperf tcmalloc and possibly might support other allocators in the future, would it be better to have a oneof here specific to the actual allocator so it is clear that the configuration only works for that allocator? I could go either way I'm just concerned that this stuff ends up being platform specific a lot and it might be better to be specific about which allocator this applies to?

Reply:
Given that the allocator is compiled in, this seems a bit finicky to me to have it be presented as runtime configurable. What do you think about if we just aborted if this is configured for an unsupported allocator since this is done at startup.

Reply:
I mean everything is compiled in at the end of the day. :)

Seems better to me to be specific about what this applies to to future proof it. If we don't do that we should be very specific about where it is supported and potentially warn/abort if it is not supported.

Reply:
Given that the allocator is compiled in
>Sorry for not being clear enough -- the allocator is effectively either tcmalloc/tcmalloc or gperftools/tcmalloc and only one of those will be compiled into the binary. What cases are you concerned about? Maybe a case where this is meant for tcmalloc allocator but is delivered to say gperftools/tcmalloc allocator?


100% agree on aborting if not supported.

Reply:
apologies for late pitch in, i agree on aborting when gperf allocator is used. Also still having hopes to understand the failed test with gperf allocator, opened a test pr here: https://github.com/envoyproxy/envoy/pull/32648/files. It can be absolutely ignored, i will be retriggering ci multiple times for `compile times options` ci check and observe raw tcmalloc stats output. Had one assumption to verify, if failing test is using huge page aware allocator to manage large memory alloc/dealloc and that releases too much memory on first run so on next round nothing is released. By looking at the impl of memory release logic in both tcmalloc and gpref tcmalloc, i could only see subtle differences, so looking at raw stats may help.